### PR TITLE
Use partial testing

### PR DESCRIPTION
**Pull request by gregorjerse**
## Checklist

* [ ] Update CHANGELOG.rst for each commit separately:
  * Pay attention to write entries under the "Unreleased" section.
  * Mark all breaking changes as "**BACKWARD INCOMPATIBLE:**" and put them
    before non-breaking changes.
  * If a commit modifies a feature listed under "Unreleased" section,
    it might be sufficient to modify the existing CHANGELOG entry from previous
    commit(s).
* [ ] Bump the process version:
  * **MAJOR version (first number)**: Backward incompatible changes (changes
    that break the api/interface). Examples: renaming the input/output, adding
    mandatory input, removing input/output...
  * **MINOR version (middle number)**: add functionality or changes in a
    backwards-compatible manner. Examples: add output field, add non-mandatory
    input parameter, use a different tool that produces same results...
  * **PATCH version (last number)**: changes/bug fixes that do not affect
    the api/interface. Examples: typo fix, change/add warning messages...
* [ ] All inputs are used in process.
* [ ] All output fields have a value assigned to them.

### Processes ###

* Code that was once accepted should be kept like it is. It should be changed
  only when its functionality has changed. Small parts may be changed
  to fix bugs (by changing code only to make it nicer you might introduce a bug).
* Set the optimal number of cores that the tool can use (check if parallelization
  can be performed).
* Description of the tool should be self explanatory with links to the tool's
  homepage and publication.
* Write explanatory input labels and add short command line parameter names
  in square brackets (for example: Maximum length [``--maxlength``]).
* Where possible provide default values for the process input fields.
  If they are provided, then ``required = False`` is not needed.
* Add descriptive error and warning messages.
* Use ``Cmd()`` for each call in the process (call of a tool, a
  Bash command, a Python script or an R script) and handle return codes.
* Add ``self.progress()`` calls where appropriate.

### Commit and PR messages ###

* Each commit should be minimal (i.e. 1 change) and self-contained (including
  tests).
* Mark incomplete PRs with the [WIP] (Work In Progress) tag.
* Do not paste links to private repositories from the public one.

### Tests ###

* Add meaningful names of test files:
  * Genomes: ``genome_species.fa.gz``
  * Annotations: ``annotation_species.gtf.gz``
  * Adapters: ``adapters-name/type-source``.
* Specific files should be saved in a folder with the name of the tested tool.
  This folder should include separate subfolders for inputs and outputs.
* Keep test files' size small. If there is no other option, add the large
  test file to the ``files/large`` directory so it will be tracked with Git LFS.
* Pay attention to edge cases: what will happen if any data is
  missing, if the length of some list is 0, if a string is given where one
  expects an integer, if a file is empty, ...
* When testing workflows, try to verify workflow success by checking one
  of the last steps. Pick the one that is most likely to fail. Also test
  that all steps have 'OK' status.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -64,7 +63,7 @@ jobs:
       # "master" branch, we allow installing pre-releases with the pip command.
       - name: Allow pre-releases for master branch
         if: github.ref == 'refs/heads/master'
-        run: echo "TOX_COMMAND=${{ env.TOX_COMMAND }} --pre" >> $GITHUB_ENV
+        run: echo "tox --pre" >> $GITHUB_ENV
```
Comment (acopar) [inline .github/workflows/ci.yml:66]: I am not sure this will be working since you removed variable assignment. You are not setting any environment variable with this and after the change the echo would not do anything. The consequence of this not working is that QA deploys would not include prerelease packages.

Comment (gregorjerse) [reply .github/workflows/ci.yml]: Nice catch. In tox 4+ it is recommended to set PIP_PRE environmental variable as --pre is deprecated and could even not work.

**Review by acopar — APPROVED** [review]

File: tox.ini [hunk 2]
```diff
@@ -25,14 +25,6 @@ passenv =
         DOCKER_*
         DJANGO_TEST_PROCESSES
 
-commands_pre =
```
Comment (dblenkus) [inline tox.ini:60]: Improve the commit message to clarify that the tox file was changed.

File: docs/CHANGELOG.rst [hunk 3]
```diff
@@ -16,6 +16,7 @@ Added
 
 Changed
 -------
+- Use partial testing when running unit tests
```
Comment (dblenkus) [inline docs/CHANGELOG.rst:19]: Improve the commit message to clarify when partial testing is used.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -23,9 +23,10 @@ jobs:
       TOX_COMMAND: "tox --skip-missing-interpreters false"
       RESOLWE_POSTGRESQL_PORT: 55433
       RESOLWE_REDIS_PORT: 56380
+      RESOLWE_TEST_ONLY_CHANGES_TO: "origin/master"
     strategy:
       matrix:
-        toxenv: ["py311", "py312"]
+        toxenv: ["py311-partial", "py312-partial"]
```
Comment (dblenkus) [inline .github/workflows/ci.yml:28]: Why do we run tests with two versions of Python? Most of the tests are for processes and thus don't make any difference, except that they take double the amount of time.

Comment (gregorjerse) [reply .github/workflows/ci.yml:28]: It's not exactly double time, but certainly double resources.

Not all tests are processes though. Some (variant tests for instance) are viewset tests. 

We could remove the `Python 3.11` tests since we use `Python 3.12` in the production.

File: docs/CHANGELOG.rst (hunk 3 above)
Comment (gregorjerse) [reply docs/CHANGELOG.rst]: Done.

File: tox.ini (hunk 2 above)
Comment (gregorjerse) [reply tox.ini:60]: Done.

**Review by dblenkus — APPROVED** [review]
